corehq/apps/users/models.py

```diff
@@ -1161,6 +1161,18 @@ def get_user_session_data(self, domain):
         })
         return session_data
 
+    def get_owner_ids(self, domain):
+        owner_ids = [self.user_id]
+        owner_ids.extend(g._id for g in self.get_case_sharing_groups(domain=domain))
+        return owner_ids
+
+    def _get_case_sharing_groups_for_locations(self, domain):
```

Comment:
Since this part of the implementation is shared, you could also make this `get_case_sharing_groups` and then in `CommCareUser` extend it, like:
```python
def get_case_sharing_groups(self, domain):
    groups = super().get_case_sharing_groups(domain)
    groups.extend(...)
    ...
```
I can think of arguments for both approaches though, so whatever you prefer sounds fine to me.